### ISSUE TITLE
Alt + select with mouse: editor looses the focus

### ISSUE THREAD

**Issue by GegznaV** [rstudio/rstudio#12310]
### System details

    RStudio Edition : Desktop [Open Source]
    RStudio Version : 2022.12.0.314
    OS Version      : Windows 10 x64 (build 19044)
    R Version       : R version 4.2.2 (2022-10-31 ucrt)

### Steps to reproduce the problem

1. Open a file in RStudio's Editor that has several rows of text.
2. Select several rows of text while pressing <kbd>Alt</kbd>
3. Cannot do anything with text (e.g., copy it), as after releasing <kbd>Alt</kbd>, the editor loses focus. It seems that the focus is moved to the "File" menu:
![image](https://user-images.githubusercontent.com/12725868/200698622-b924e3d9-29b1-42f9-9bdb-afd331a26391.png)

### Describe the problem in detail

<kbd>Alt</kbd> + Selecting several rows is inefficient now as it is not possible to do anything with the selected text, as the focus is lost.

### Describe the behavior you expected

I expect the same behavior as in the current stable RStudio version 2002.07.2 while after releasing <kbd>Alt</kbd> the focus is not moved/lost.

**Comment by ronblum** [comment]
@GegznaV Does it work the first time you click on Alt, but then focus switches to the "File" menu? That's what I'm seeing both in 2022.07.2+576. And then clicking on Alt again returns focus. It's also typical of other applications, such as Notepad, WordPad, and Slack.

**Comment by GegznaV** [comment]
Focus is lost after <kbd>Alt</kbd> is released. No issue with this in Notepad++, VSCode, RStudio 2022.07.2, but there's a problem in RStudio daily builds (now I have 2022.12.0.318). 

You may notice in the gif, that <kbd>Delete<kbd> first time has no effect on the selection:

![RS-alt-isue](https://user-images.githubusercontent.com/12725868/200899886-6caee6d8-d673-4c0c-8665-0c9859d29d55.gif)

After pressing <kbd>Alt</kbd> once more it seems that the focus is back. But it is inconvenient as for operations that have no visible effects such as <kbd>Ctrl<kbd>+<kbd>C<kbd>  it is not clear if the action was performed successfully or not as extra attention should be paid where the focus is. It was not the case in the previous RStudio version.

It seems that some new functionality clashes with the old functionality.

Did any new option appear in RStudio, that enabled <kbd>Alt<kbd> to change focus?

**Comment by GegznaV** [comment]
@ronblum, can you reproduce the `Alt(hold) → Select → Alt(release) → Delete → Nothing happens` sequence or behavior is different for you?

Can you confirm that the behavior is the same on both stable and daily builds RStudio versions for you?

**Comment by rjvelasquezm** [comment]
FWIW I see this same behavior (described here https://github.com/rstudio/rstudio/issues/12231#issuecomment-1303557929) when testing the electron build.

**Comment by GegznaV** [comment]
The issue is that in the new version of RStudio, clicking `Alt` automatically selects `File` menu in the main menu bar (next, you can use `Tab` key to navigate to the following menu options, or e.g., `Alt` + `F` to expand files menu, find gif below). Is there an option to suppress this new behaviour of `Alt`?
![RS-alt-new-behavior](https://user-images.githubusercontent.com/12725868/207617874-d0c7746e-215f-41a5-8db7-54dcc15b463b.gif)

@ronblum , is the new behaviour of the `Alt` button different  with RStudio 2022.12.0+353 Elsbeth Geranium (desktop) on your computer than the one shown in the GIF and described above as you added the label that `repro needed`? Does `repro needed` mean that this behaviour is unique to my computer only?

I have just noticed, that the behaviour I want (not to lose focus) can also be achieved by clicking and holding `Alt` + `Shift`. But it is confusing, as by clicking and holding `Alt` a cross-hair appears and that indicates that the same behaviour could be achieved just by holding `Alt`. So the other option could be to remove the multiline selection option when only `Alt` is held (this is the case in, e.g., VSCode: here only `Alt` + `Shift` allows choosing multiple rows from the same row position).

**Comment by kevinushey** [comment]
Just to confirm, with this sequence of actions: "press and holt Alt, click the mouse to make a multi-cursor selection; release Alt"

- On older builds of RStudio, focus is not sent to the main menu bar;
- On new (Electron) builds of RStudio, focus is sent to the main menu bar.

In both builds, pressing Alt alone does send focus to the main menu bar (as expected).

As I understand it, clicking the mouse while Alt is held down should "cancel" the "send focus to main menu bar" action, but that's not happening here.

**Comment by kevinushey** [comment]
This sort of feels like an Electron bug; it would be worth testing if this reproduces in the builds of RStudio using the latest version of Electron.

**Comment by ronblum** [comment]
@kevinushey and @timtmok it looks like this is an Electron bug: https://github.com/electron/electron/issues/34211. There's also a description of how the behavior was adjusted in VS Code: https://github.com/electron/electron/issues/34211#issuecomment-1182181107.

**Comment by kevinushey** [comment]
Triaging as we potentially have a low-cost fix as described by @ronblum in https://github.com/rstudio/rstudio/issues/12310#issuecomment-1422960047 (thanks!)

**Comment by GegznaV** [comment]
It would be nice toa have this fixed.

**Comment by mikebessuille** [comment]
Looks like a duplicate of https://github.com/rstudio/rstudio/issues/12470

**Comment by MintyFreshFarmer** [comment]
Is this problem fixed? I'm on 2023.03.0+386 and this problem popped up. The only bypass is this sequence: ctrl -> alt -> press shift (or other buttons, so far just backpsace and s, don't do escape) and let go -> left click and place cursor where you want it. 

It's weird, but I found this after some testing for key glitches. At some point in time I was able to use shift+alt to get multi cursor to work in this version, but somehow it disappeared. Very bugged. Hope they fix it, multi cursor is a staple function lol.

**Comment by douglaswhitaker** [comment]
I'm still having this issue with the most recent version of RStudio, but the Shift + Alt suggestion earlier in the thread seems to be working (at least for the first day I've tried it).

RStudio 2023.06.0+421 "Mountain Hydrangea" Release (583b465ecc45e60ee9de085148cd2f9741cc5214, 2023-06-05) for windows
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2023.06.0+421 Chrome/110.0.5481.208 Electron/23.3.0 Safari/537.36

**Comment by rjvelasquezm** [comment]
Curious if there are any plans to fix this?

Personally, and Im sure others have the issue, it just doesnt make sense to update to the latest RStudios where my workflow will take a hit unless its just never going to happen and Ill just have to accept it and deal with it.

**Comment by timtmok** [comment]
This behaviour comes from Electron: https://github.com/electron/electron/issues/34211

Implementing #8932 might work around the issue as some have reported Electron apps that hide the menu bar will no longer immediately change focus. Short of disabling the `Alt` behaviour with some custom handling and logic, this is likely to wait on an Electron change. Any change here must ensure that keyboard navigation can activate menus while still allowing `Alt` to column select text in the editor.

**Comment by Saviouress** [comment]
This bug is so annoying that I'm still on RStudio 2022.07.2

**Comment by GegznaV** [comment]
> This bug is so annoying that I'm still on RStudio 2022.07.2

Try using <kbd>Shift</kbd>+<kbd>Alt</kbd> instead of just <kbd>Alt</kbd>. It took a while for me to learn the new habit, but it allows using the current version of RStudio.

**Comment by RRTogunov** [comment]
For those on Windows, another (very Jankey) workaround using Autohotkey is to trigger "right click" and "escape" whenever you press click+ alt
Took me forever to figure out something that works and doesn't interfere with other functions (eg multicursor). 
Here is code using AutoHotKey V2 
```
; multi-line select
    #HotIf WinActive("ahk_exe rstudio.exe")
    ~LButton & LAlt::{
        KeyWait "LButton"
        KeyWait "LAlt"
        send "{RButton}{Esc}"
      }
    #HotIf
```
#HotIf WinActive: Triggers only in rstudio
~: permits native function of click+alt (i.e., doesn't hide input from system)
KeyWait: waits for mouse and alt to be released
send: send right click followed by escape (For rare occasions when rstudio does not lose focus, sending {alt} would inadvertently shift focus to menu bar and {esc} would cancel multiline selection)

also, applicable to duplicate issues:
https://github.com/rstudio/rstudio/issues/12470#issue-1505031998
https://github.com/rstudio/rstudio/issues/12906#issue-1640254786

**Comment by max-nothacker** [comment]
This bug wasn't fixed yet, was it?

**Comment by Zweckoptimismus** [comment]
No, unfortunately not. My workflow has now changed to "release Alt key and press Alt key again" to unfocus the menu and instead return the focus to script window. :-/
